src/renderer/services/tokens.js

```diff
@@ -8,16 +8,28 @@ export default {
     if (this.tokenExists(data.assetId, data.network)) {
       const existing = this.getOne(data.assetId, data.network);
       if (existing) {
+        let skipUpdate = false;
         if (existing.isCustom === data.isCustom
           || (existing.isCustom === true && data.isCustom === false)) {
-          return this;
+          // skip updating if already in local db and isCustom is the same
+          //   or isCustom is already true and this would set it back to false
+          skipUpdate = true;
+        }
+
+        if (data.sale || existing.sale) {
```

Comment:
This should allow the server to overwrite it to add or remove the sale, so that looks good. Should it be preserving the isCustom flag though if existing had it?